addons/OPT/GPS/fn_clientInit.sqf

```diff
@@ -1,120 +1,60 @@
-Step 1 initializes _units and _vehicles with what needs to be drawn (init), step 2 draws a single icon (addIcons).
-*/
-GVAR(iconStatemachine) = call CFUNC(createStatemachine);
-
-[GVAR(iconStatemachine), "init", {
-    private _units = allUnits select {[_x] call FUNC(isUnitVisible)};
-
-    // TODO: For now, we ignore empty vehicles
-    // private _vehicles = vehicles select {[_x] call FUNC(isVehicleVisible)};
-    private _vehicles = [];
-
-    // Figure out what icons we had last time. Maybe it's different this time around and we need to remove some icons!
-    GVAR(lastProcessedIcons) = (CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0};
-
-    {
-        DUMP("ICON REMOVED: " + _x);
-        [_x, "hoverin"] call CFUNC(removeMapGraphicsEventHandler);
-        [_x, "hoverout"] call CFUNC(removeMapGraphicsEventHandler);
-        [_x] call CFUNC(removeMapGraphicsGroup);
-    } count (GVAR(lastProcessedIcons) - GVAR(processedIcons));
-
-    // Now we got a clean slate again, let's start the cycle.
-    GVAR(processedIcons) = [];
-
-    if (_units isEqualTo [] && _vehicles isEqualTo []) then {
-        "init"
-    } else {
-        ["addIcons", [_units, _vehicles]]
+    // Create markers for all players already here in case of JIP
+    allPlayers apply {
+        DUMP(_x);
+        DUMP([_x] call FUNC(isUnitVisible));
+        if (CLib_Player != _x && [_x] call FUNC(isUnitVisible) ) then {
+            DUMP(format ["UNIT ICON ADDED: %1", _x]);
+            [_x] call FUNC(addUnitToGPS);
+            DUMP((CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0});
+        };
     };
-}] call CFUNC(addStatemachineState);
-
-[GVAR(iconStatemachine), "addIcons", {
-    params ["", "_data"];
-    _data params ["_units", "_vehicles"];
-
-    // First, we process a single Unit
-    if (!(_units isEqualTo [])) then {
-        private _unit = _units deleteAt 0;
 
-        // If it's an invalid unit, move on to the next one
-        while {
-            !([_unit] call FUNC(isUnitVisible))
-            && !([CLib_Player] call FUNC(isUnitLeader))
-            && group _unit != group CLib_Player
-            && !([_unit] call FUNC(isUnitLeader))
-            && _unit getVariable ["FAR_isUnconscious", 0] == 0
-            && !(_units isEqualTo [])
-        } do {
-            _unit = _units deleteAt 0;
+    // We have to register the UAV dialog every time the map is opened/closed
+    addMissionEventHandler ["Map", {
+        params ["_mapIsOpened", "_mapIsForced"];
+        DUMP("Map opened");
+
+        if (_mapIsOpened) then {
+            // UAV
+            if (!isNull (findDisplay 160 displayCtrl 51)) exitWith {
+                DUMP("UAV dialog found");
+                DUMP((findDisplay 160) displayCtrl 51);
+                (findDisplay 160 displayCtrl 51) call CFUNC(registerMapControl);
+            };
         };
```

Comment:
Wenn du das aktuelle Map-Control in einer globalen Variable speicherst, kannst du es hier in einem `else`-Block auch wieder aus CLibs Cache löschen. Dann wächst das interne CLib-Array nicht ins unendliche...